apm-agent-core/src/main/java/co/elastic/apm/report/serialize/DslJsonSerializer.java

```diff
@@ -564,20 +564,37 @@ private boolean isLibraryFrame(String className) {
     private void serializeSpanContext(SpanContext context) {
         writeFieldName("context");
         jw.writeByte(OBJECT_START);
-        writeFieldName("db");
-        jw.writeByte(OBJECT_START);
-        final Db db = context.getDb();
-        writeField("instance", db.getInstance());
-        writeLongStringField("statement", db.getStatement());
-        writeField("type", db.getType());
-        writeField("user", db.getUser());
-        writeFieldName("tags");
-        serializeTags(context.getTags());
-        jw.writeByte(OBJECT_END);
+
+        boolean dbContextWritten = serializeDbContext(context.getDb());
+
+        Map<String, String> tags = context.getTags();
+        if (!tags.isEmpty()) {
+            if(dbContextWritten)
```

Comment:
Currently, methods like `serializeDbContext` add the comma themselves when they are written and the last object, like tags, is always written, even if it's empty. That makes sure the comma is not the last thing in an object.

Reply:
meaning - you want to always add the `tags` section, even if it mostly empty?

Reply:
That would make the code more concise but on the other hand, increases the size of the JSON unnecessarily. I'm torn between the two options... What do you think?

Reply:
I would say not adding by default a section that is normally empty. Serialization code is like that sometimes...

Reply:
Ok, I'm good with that. Let's leave it as-is.